server/src/main/java/org/elasticsearch/common/util/CuckooFilter.java

```diff
@@ -122,17 +124,26 @@
 
         this.fingerprintMask = (0x80000000 >> (bitsPerEntry - 1)) >>> (Integer.SIZE - bitsPerEntry);
 
-        data = (PackedInts.Mutable) PackedInts.getReader(new DataInput() {
-            @Override
-            public byte readByte() throws IOException {
-                return in.readByte();
-            }
-
-            @Override
-            public void readBytes(byte[] b, int offset, int len) throws IOException {
-                in.readBytes(b, offset, len);
+        if (in.getVersion().before(Version.V_8_0_0)) {
+            final PackedInts.Reader reader = PackedInts.getReader(new DataInput() {
```

Comment:
`PackedInts.getReader()` doesn't exist in lucene 9, so is the plan to remove this further in a follow-up or do we need to rework it here?

Reply:
This will be removed once we backport the change to 7.x